internal/common/cloudproviderapi/client.go

```diff
@@ -46,108 +58,159 @@ type AWSAccount struct {
+	path := fmt.Sprintf("/api/v2/stacks/%s/aws/jobs/cloudwatch/%s", stackID, jobName)
+	respData := apiResponseWrapper[AWSCloudWatchScrapeJob]{}
+	err := c.doAPIRequest(ctx, http.MethodGet, path, nil, &respData)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get AWS CloudWatch scrape job: %w", err)
+	}
+	return &respData.Data, nil
+}
+
+func (c *Client) ListAWSCloudWatchScrapeJobs(ctx context.Context, stackID string) ([]AWSCloudWatchScrapeJob, error) {
+	path := fmt.Sprintf("/api/v2/stacks/%s/aws/jobs/cloudwatch", stackID)
+	respData := apiResponseWrapper[[]AWSCloudWatchScrapeJob]{}
+	err := c.doAPIRequest(ctx, http.MethodGet, path, nil, &respData)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get AWS CloudWatch scrape job: %w", err)
+	}
+	return respData.Data, nil
+}
+
+func (c *Client) UpdateAWSCloudWatchScrapeJob(ctx context.Context, stackID string, jobName string, jobData AWSCloudWatchScrapeJob) (*AWSCloudWatchScrapeJob, error) {
+	path := fmt.Sprintf("/api/v2/stacks/%s/aws/jobs/cloudwatch/%s", stackID, jobName)
+	respData := apiResponseWrapper[AWSCloudWatchScrapeJob]{}
+	err := c.doAPIRequest(ctx, http.MethodPut, path, &jobData, &respData)
+	if err != nil {
+		return nil, fmt.Errorf("failed to update AWS CloudWatch scrape job: %w", err)
+	}
+	return &respData.Data, nil
+}
+
+func (c *Client) DeleteAWSCloudWatchScrapeJob(ctx context.Context, stackID string, jobName string) error {
+	path := fmt.Sprintf("/api/v2/stacks/%s/aws/jobs/cloudwatch/%s", stackID, jobName)
+	err := c.doAPIRequest(ctx, http.MethodDelete, path, nil, nil)
+	if err != nil {
+		return fmt.Errorf("failed to delete AWS CloudWatch scrape job: %w", err)
+	}
+	return nil
+}
+
+func (c *Client) doAPIRequest(ctx context.Context, method string, path string, body any, responseData any) error {
 	var reqBodyBytes io.Reader
 	if body != nil {
 		bs, err := json.Marshal(body)
 		if err != nil {
-			return nil, fmt.Errorf("failed to marshal request body: %w", err)
+			return fmt.Errorf("failed to marshal request body: %w", err)
 		}
 		reqBodyBytes = bytes.NewReader(bs)
 	}
 	var resp *http.Response
 
 	req, err := http.NewRequestWithContext(ctx, method, c.apiURL.String()+path, reqBodyBytes)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create request: %w", err)
+		return fmt.Errorf("failed to create request: %w", err)
 	}
 	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", c.authToken))
 	req.Header.Add("Content-Type", "application/json")
 
 	resp, err = c.client.Do(req)
 	if err != nil {
-		return nil, fmt.Errorf("failed to do request: %w", err)
+		return fmt.Errorf("failed to do request: %w", err)
 	}
 
 	bodyContents, err := io.ReadAll(resp.Body)
 	resp.Body.Close()
 	if err != nil {
-		return nil, fmt.Errorf("failed to read response body: %w", err)
+		return fmt.Errorf("failed to read response body: %w", err)
 	}
 	if resp.StatusCode >= 400 {
```

Comment:
should we instead catch in here:

```suggestion
	if !(resp.StatusCode >= 200 && resp.StatusCode <= 299 ) {
```

just a nit, in case for some reason a gw starts responding some weird, and we end up going to the following conditional and unmarshalling some weird data.